test/functional/testSettings.js

```diff
@@ -0,0 +1,740 @@
+       */
+      'testSettingsInitialState': function () {
+        var remote = this.remote;
+        var testSubName = 'testSettingsInitialState';
+
+        return settingsPage
+          .getTimeBasedEventsCheckbox()
+          .isSelected()
+          .then(function (selected) {
+            expect(selected).to.be.ok();
+          })
+          .then(function () {
+            return settingsPage
+              .getNameIsPatternCheckbox()
+              .isSelected()
+              .then(function (nameIsPatternSelected) {
+                expect(nameIsPatternSelected).to.not.be.ok();
+              });
+          })
+          .then(function () {
+            return settingsPage
+              .getIndexPatternField()
+              .then(function (patternField) {
+                return patternField
+                  .getProperty('value')
+                  .then(function (pattern) {
+                    common.log('patternField value = ' + pattern);
+                    expect(pattern).to.be('logstash-*');
+                  });
+              });
+          })
+          .then(function () {
+            return settingsPage.getTimeFieldNameField()
+              .isSelected()
+              .then(function (timeFieldIsSelected) {
+                common.log('timeField isSelected = ' + timeFieldIsSelected);
+                expect(timeFieldIsSelected).to.not.be.ok();
+              });
+
+          })
+          .catch(function screenshotError(reason) {
+            common.log('Test Failed, taking screenshot "./screenshot-' + testSubName + '-ERROR- + Date.now() + .png"');
+            return remote
+              .takeScreenshot()
+              .then(function screenshot2a(data) {
+                fs.writeFileSync('./screenshot-' + testSubName + '-ERROR-' + Date.now() + '.png', data);
+                throw new Error(reason);
+              });
+          });
+      },
+
+      /*
+       ** Ensure Create button is disabled until you select a time field
+       */
+      'testCreateButtonDisabledUntilTimeFieldSelected': function () {
+        var testSubName = 'testCreateButtonDisabledUntilTimeFieldSelected';
+        var remote = this.remote;
+
+        return settingsPage
+          .getCreateButton()
+          .isEnabled()
+          .then(function (enabled) {
+            expect(enabled).to.not.be.ok();
+          })
+          .then(function () {
+            // select a time field and check that Create button is enabled
+            return settingsPage
+              .selectTimeFieldOption('@timestamp');
+          })
+          .then(function () {
+            return settingsPage
+              .getCreateButton()
+              .isEnabled()
+              .then(function (enabled) {
+                expect(enabled).to.be.ok();
+              });
+          })
+          .catch(function screenshotError(reason) {
+            common.log('Test Failed, taking screenshot "./screenshot-' + testSubName + '-ERROR- + Date.now() + .png"');
```

Comment:
Thanks for the feedback!  Yes, I was initially a bit lazy on the log message and it was really logging 'Date.now()' because I didn't want to the timestamp of the log not to match the timestamp of the filename.  But now I saved the time and use it in both the message and the filename.  And I moved it to Common.js.